src/xwayland.c

```diff
@@ -386,6 +386,13 @@ set_activated(struct view *view, bool activated)
 	wlr_xwayland_surface_activate(surface, activated);
 	if (activated) {
 		wlr_xwayland_surface_restack(surface, NULL, XCB_STACK_MODE_ABOVE);
+		/* Restack unmanaged surfaces on top */
+		struct xwayland_unmanaged *u;
+		struct wl_list *list = &view->server->unmanaged_surfaces;
+		wl_list_for_each (u, list, link) {
+			wlr_xwayland_surface_restack(u->xwayland_surface, NULL,
+						     XCB_STACK_MODE_ABOVE);
```

Comment:
I think we need to check for `u->node != NULL` here.

Other than that: LGTM

Reply:
Good call